### PR TITLE
Wilson/fullscreen payment modal

### PR DESCRIPTION
**Pull request by wilsonj806**
See attached [Notion page](https://www.notion.so/Fullscreen-payment-flow-a3936a2115ef4817aca35b04de4f365f) for screens.

Changes:
- Make modal full screen
- Update init GAM modal so that the steps diagram is hidden on mobile
- Add progress bar to int GAM modal
- Update init donation pool  modal so that donation presets are rendered in a 2x2 grid on mobile screens
- Update credit card modal so CCV/ Expiry/ Zip Code are rendered inline with the credit card number on desktop
- Add scroll to top to several modal screens

### CODE REVIEW

File: src/components/ModalPayment/ModalBuyMeal/ModalBuyMealPool.tsx [hunk 1]
```diff
@@ -59,17 +62,22 @@ export const Modal = (props: Props) => {
...
   }, []);
 
-  const fetchData = async (sellerId: string) => {
+  const fetchData = async (sellerId: string, campaignId?: string) => {
     // Note(wilsonj806) Showing the campaign that expires first
     // will need to update this if we render multiple distributors
     const { data } = await getCampaignsForMerchant(sellerId);
-    const { data: distrib } = await getDistributor(data[0].distributor_id);
+
+    const campaign = data.find((ele) => campaignId === ele.id);
```
Comment (aliceluo2048) [inline src/components/ModalPayment/ModalBuyMeal/ModalBuyMealPool.tsx:70]: Why is `campaignId` optional and wouldn't this blow up if it doesn't get passed in?

File: src/components/ModalPayment/ModalAmount/ModalAmount.tsx [hunk 2]
```diff
@@ -232,3 +234,5 @@ const Header = styled.div`
   font-size: 30px;
   font-weight: 600;
 `;
+
+ContentContainer.displayName = 'Donation Pool Amount Form';
```
Comment (aliceluo2048) [inline src/components/ModalPayment/ModalAmount/ModalAmount.tsx:238]: Is this needed?

Comment (wilsonj806) [reply src/components/ModalPayment/ModalAmount/ModalAmount.tsx:238]: Missed it, will delete

File: src/components/ModalPayment/ModalAmount/ModalAmount.tsx [hunk 3]
```diff
@@ -72,7 +72,7 @@ export const Modal = (props: Props) => {
   };
 
   return (
-    <ContentContainer id="donation-form" data-testid="modal-amount">
+    <ContentContainer>
```
Comment (ath-chen) [inline src/components/ModalPayment/ModalAmount/ModalAmount.tsx:75]: why is data-testid removed? is there no testing associated w that?

File: src/components/ModalPayment/ModalBuyMeal/ModalBuyMealPool.tsx (hunk 1 above)
Comment (wilsonj806) [reply src/components/ModalPayment/ModalBuyMeal/ModalBuyMealPool.tsx:70]: The `useEffect` that it's called in on line 77 addresses that, but I'll try fix it so it looks less jank.

File: src/components/ModalPayment/ModalAmount/ModalAmount.tsx (hunk 3 above)
Comment (wilsonj806) [reply src/components/ModalPayment/ModalAmount/ModalAmount.tsx:75]: I actually don't know and should have checked, will revert.

**Review by ath-chen — APPROVED** [review]

**Pull request merged**